src/app/3d/qgs3dmapcanvas.cpp

```diff
@@ -89,3 +95,19 @@ void Qgs3DMapCanvas::setViewFromTop( const QgsPointXY &center, float distance, f
   float worldY = center.y() - mMap->origin().y();
   mScene->cameraController()->setViewFromTop( worldX, -worldY, distance, rotation );
 }
+
+void Qgs3DMapCanvas::saveAsImage( const QString fileName, const QString fileFormat )
+{
+  if ( !fileName.isEmpty() )
+  {
+    mCaptureReply = mCapture->requestCapture();
```

Comment:
mCaptureReply probably does not need to be a member variable of the class - just a local variable. Also in theory there could be more than one pending capture request...

Reply:
You are right.